### PR TITLE
Junhyung getactivequarterlist component

### LINKED ISSUE #8: Implement `getActiveQuarterList` method (Backend only)
As a precursor to a future new job that updates all course information for all active quarters at some scheduled interval (e.g. once an hour, once a day, etc.) in this issue you will add a new method to `UCSBAPIQuarterService` called `getActiveQuarterList`  that will return an `ArrayList<String>` value that is a list of the YYYYQ codes starting with the current quarter, and ending with the current value returned by getEndQtrYYYYQ.

You'll also add a endpoint to the `UCSBAPIQuarterController` as follows:

`GET /api/public/activeQuarters` 

the invokes this method and returns the value.

# Details

The `UCSBAPIQuarterService` has these methods:

* `public String getStartQtrYYYYQ()` returns the start quarter for the app in YYYYQ format (i.e. the oldest quarter that is available in the drop down menus in the app).
*  `public String getEndQtrYYYYQ()` returns the start quarter for the app in YYYYQ format (i.e. the newest quarter that is available in the drop down menus in the app, which may be in the future)
* `public String getCurrentQuarterYYYYQ()` gets the current quarter in YYYYQ format.

There is also this method in `models/Quarter.java`

 * `public static List<Quarter> quarterList(String start, String end)` which returns a list of Quarter objects starting at start quarter (specified in YYYYQ format) and ending with end quarter  (specified in YYYYQ format)

Using these tools, and *not* reinventing the wheel, but making as much use of existing methods as possible, add a method to `UCSBAPIQuarterService` called `getActiveQuarterList`  that will return an `ArrayList<String>` value that is a list of the YYYYQ codes starting with the current quarter, and ending with the current value returned by getEndQtrYYYYQ.

For example:

* If the value returned by `getCurrentQuarterYYYYQ` is `20244`  (F24) and the value of END_QTR is `20251` (W25), then the value returned by `getActiveQuarterList` should be `["20244", "20251"]`.
* If the value  returned by `getCurrentQuarterYYYYQ` is `20252`  (S25) and the value of END_QTR is `20254` (F25), then the value returned by `getActiveQuarterList` should be `["20252", "20253", "20254"]`.

# Acceptance Criteria:

- [x] New method implemented as described above
- [x] New controller endpoint implemented as described above

### PR DESCRIPTION
**Pull request by Neamal**
Closes #18 

Dokku Deployment: https://courses-dev-neamal.dokku-01.cs.ucsb.edu/#

Finished the getActiveQuarterList method in the UCSBApiQuarterService file and created an endpoint called activeQuarters in the controllers file. Also created tests for both the getActiveQuarterList method as well as the endpoint. Passed all of the tests including coverage and stryker. Below is a picture of the component in Swagger as it lists all of the dates until 20244 which was listed in my test file.

image

### CODE REVIEW

**Review by vincentc0202 — DISMISSED** [review]
Good concise tests and good job adding the backend part

File: src/test/java/edu/ucsb/cs156/courses/controllers/UCSBAPIQuarterControllerTests.java [hunk 1]
```diff
@@ -111,4 +111,40 @@ public void test_loadQuarters() throws Exception {
...
+
+  //   @Test
+  //   public void test_getActiveQuarterList_F19_S20() throws Exception {
+  //     List<Quarter> expected = new ArrayList<Quarter>();
+  //     expected.add(new Quarter("F19"));
+  //     expected.add(new Quarter("W20"));
+  //     expected.add(new Quarter("S20"));
+
+  //     assertEquals(expected, Quarter.quarterList("20194", "20202"));
+  //   }
```
Comment (pconrad) [inline src/test/java/edu/ucsb/cs156/courses/controllers/UCSBAPIQuarterControllerTests.java]: https://ucsb-cs156.github.io/topics/code_reviews/commented_out_code.html#why-should-commented-out-code-be-removed

**Review by pconrad — CHANGES_REQUESTED** [review]
https://ucsb-cs156.github.io/topics/code_reviews/commented_out_code.html#why-should-commented-out-code-be-removed

**Review by pconrad — APPROVED** [review]
Code LGTM.. but need to test with the setting of END_QTR in the future.

**Comment by pconrad** [comment]
@Neamal this looks good, and I think we can get it merged soon.  Just one thing: can you do this on your dokku instance?

```
dokku config:set courses-dev-neamal END_QTR=20251
```

It should restart your app after you do this.  This is needed because otherwise, the endpoint will "count backwards" from the current quarter to the END_QTR, which is not what we want.   We need to test the app with values that are realistic.

Then I think we can get this one merged.

**Pull request merged**